src/librustc_mir/interpret/operand.rs

```diff
@@ -608,6 +608,13 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                 } else {
                     self.param_env
                 };
+                // We use `const_eval` here and `const_eval_raw` elsewhere in mir interpretation.
+                // The reason we use `const_eval_raw` everywhere else is to prevent cycles during
+                // validation, because validation automatically reads through any references, thus
+                // potentially requiring the current static to be evaluated again. This is not a
+                // problem here, because we need an operand and operands are always reads.
```

Comment:
I don't understand the last sentence. "operands are always reads"? Previously the comment explains that reads are exactly the problem.

Reply:
What I'm trying to say is that if we are going to read immediately after this function, we may as well read right here.

Reply:
Ah I see. Not sure if we always do that for operands though... we might create it and never read from it, or we might project into it before reading.

Reply:
Well... semantically an operand is a read. It's true that there may be cases where we don't read and just project and then read a zst or sth. I'll see if const eval can notice this difference

Reply:
>  semantically an operand is a read

I have a hard time parsing your statement,
Do you mean *creating* an operand? If so, I tend to agree, though that's not what we actually implement. But in terms of UB detection we likely should. That's a separate concern from this PR though.

Reply:
Yes, that's what I tried to convey

Reply:
There's still no explanation though why this PR changes from `_raw` to non-raw.

Reply:
I have forgotten what it is. Something with #67000 otherwise changing diagnostics.

Reply:
Ah well, I guess I'm just going to trust you on this then.

Reply:
Yea, I'm sorry about the untracable change reason. I'm trusting myself that my past self had a good reason here, too.